src/main/kotlin/no/nav/amt/deltaker/bff/deltaker/DeltakerV2Consumer.kt

```diff
@@ -0,0 +1,118 @@
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.kafka.common.serialization.UUIDDeserializer
+import org.slf4j.LoggerFactory
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.UUID
+
+class DeltakerV2Consumer(
+    private val deltakerService: DeltakerService,
+    kafkaConfig: KafkaConfig = if (Environment.isLocal()) LocalKafkaConfig() else KafkaConfigImpl(),
+) : Consumer<UUID, String?> {
+
+    private val log = LoggerFactory.getLogger(javaClass)
+
+    private val consumer = ManagedKafkaConsumer(
+        topic = Environment.AMT_DELTAKERV2_TOPIC,
+        config = kafkaConfig.consumerConfig(
+            keyDeserializer = UUIDDeserializer(),
+            valueDeserializer = StringDeserializer(),
+            groupId = Environment.KAFKA_CONSUMER_GROUP_ID,
+        ),
+        consume = ::consume,
+    )
+
+    override suspend fun consume(key: UUID, value: String?) {
+        if (value == null) {
+            log.info("Mottok tombstone for deltaker $key - håndterer ikke meldingen")
+            return
+        }
+
+        val deltakerV2 = objectMapper.readValue<DeltakerV2Dto>(value)
+
+        if (deltakerV2.kilde != DeltakerV2Dto.Kilde.KOMET) return
+
+        deltakerService.oppdaterDeltaker(deltakerV2.toDeltakerOppdatering())
+    }
+
+    override fun run() = consumer.run()
+}
+
+data class Deltakeroppdatering(
+    val id: UUID,
+    val startdato: LocalDate?,
+    val sluttdato: LocalDate?,
+    val dagerPerUke: Float?,
+    val deltakelsesprosent: Float?,
+    val bakgrunnsinformasjon: String?,
+    val innhold: List<Innhold>,
+    val status: DeltakerStatus,
+    val historikk: List<DeltakerHistorikk>,
+)
+
+data class DeltakerV2Dto(
+    val id: UUID,
+    val status: DeltakerStatusDto,
+    val dagerPerUke: Float?,
+    val prosentStilling: Double?,
+    val oppstartsdato: LocalDate?,
+    val sluttdato: LocalDate?,
+    val bestillingTekst: String?,
+    val kilde: Kilde?,
+    val innhold: Deltakelsesinnhold?,
+    val historikk: List<DeltakerHistorikk>?,
+) {
+    fun toDeltakerOppdatering(): Deltakeroppdatering {
+        require(status.id != null) { "Kan ikke håndtere deltakerstatus uten id for deltaker: $id, kilde: $kilde" }
+
+        return Deltakeroppdatering(
+            id = id,
+            startdato = oppstartsdato,
+            sluttdato = sluttdato,
+            dagerPerUke = dagerPerUke,
+            deltakelsesprosent = prosentStilling?.toFloat(),
+            bakgrunnsinformasjon = bestillingTekst,
+            innhold = innhold?.innhold ?: emptyList(),
+            status = DeltakerStatus(
+                id = status.id,
+                type = status.type,
+                aarsak = status.aarsak,
+                gyldigFra = status.gyldigFra,
+                gyldigTil = null,
+                opprettet = status.opprettetDato,
+            ),
+            historikk = historikk ?: emptyList(),
```

Comment:
Denne listen skal ikke kunne være tom, for man må som minimum ha et vedtak som ikke er fattet der (det som blir opprettet når man deler utkastet :))